lib/LedgerNodePeers.js

```diff
@@ -0,0 +1,147 @@
+        updated: now
+      },
+      peer,
+    };
+    let result;
+    try {
+      result = await collection.insertOne(peerRecord);
+    } catch(e) {
+      if(database.isDuplicateError(e)) {
+        throw new BedrockError(
+          'Duplicate peer.',
+          'DuplicateError', {
+            httpStatusCode: 409,
+            public: true,
+            ledgerNodeId: this.ledgerNodeId,
+            peerId: peer.id,
+          }, e);
+      }
+      // throw all other error types
+      throw e;
+    }
+
+    return result.ops[0];
+  }
+
+  async get({id}) {
+    const collection = database.collections[COLLLECTION_NAME];
+    const query = {
+      'meta.ledgerNodeId': this.ledgerNodeId,
+      'peer.id': id,
+    };
+    const result = await collection.findOne(query, {
+      projection: {_id: 0, peer: 1}
+    });
+
+    if(!result) {
+      throw new BedrockError('Peer not found.', 'NotFoundError', {
+        httpStatusCode: 404,
+        ledgerNodeId: this.ledgerNodeId,
+        peerId: id,
+        public: true,
+      });
+    }
+
+    return result.peer;
+  }
+
+  async getAll() {
+    const collection = database.collections[COLLLECTION_NAME];
+    const query = {
+      'meta.ledgerNodeId': this.ledgerNodeId,
+    };
+    const result = await collection.find(query, {
+      projection: {_id: 0, peer: 1}
+    }).toArray();
+
+    const peers = result.map(r => r.peer);
+
+    return peers;
+  }
+
+  async getRecommended() {
+    const collection = database.collections[COLLLECTION_NAME];
+    const query = {
+      'meta.ledgerNodeId': this.ledgerNodeId,
+      'peer.recommended': true,
+    };
+    const result = await collection.find(query, {
+      projection: {_id: 0, peer: 1}
+    }).toArray();
+
+    const peers = result.map(r => r.peer);
+
+    return peers;
+  }
+
+  // peers are permanently deleted, not marked as deleted
+  async remove({id}) {
+    const collection = database.collections[COLLLECTION_NAME];
+    const query = {
+      'meta.ledgerNodeId': this.ledgerNodeId,
+      'peer.id': id,
+    };
+    const result = await collection.deleteOne(query);
+
+    if(result.deletedCount === 0) {
+      throw new BedrockError('Peer not found.', 'NotFoundError', {
+        httpStatusCode: 404,
+        ledgerNodeId: this.ledgerNodeId,
+        peerId: id,
+        public: true,
+      });
+    }
+
+    return {success: true};
+  }
+
+  async update({peer}) {
+    const collection = database.collections[COLLLECTION_NAME];
+
+    const query = {
+      'peer.id': peer.id,
+      'meta.ledgerNodeId': this.ledgerNodeId,
+    };
+
+    const result = await collection.updateOne(query, {
+      $set: {peer, 'meta.updated': Date.now()}
+    });
+
+    if(result.matchedCount !== 1) {
+      throw new BedrockError('Peer not found.', 'NotFoundError', {
+        httpStatusCode: 404,
+        ledgerNodeId: this.ledgerNodeId,
+        peerId: peer.id,
+        public: true,
+      });
+    }
+
+    return result.result;
```

Comment:
@JoshDunnDev @JSAssassin @mandyvenables @aljones15 This is a good example, when dealing with the mongo APIs it can be confusing because there is the documentation for the mongoAPIs that Andrew linked to above, and then there is the documentation for the Node.js mongo driver we use. The driver API is our source of truth.

https://mongodb.github.io/node-mongodb-native/3.5/api/Collection.html#updateOne
https://mongodb.github.io/node-mongodb-native/3.5/api/Collection.html#~updateWriteOpCallback